src/app/(private)/write/(components)/publish-article-dialog.tsx

```diff
@@ -46,14 +47,21 @@ export const PublishArticleDialog = ({ children }: PropsWithChildren) => {
     }
   }, [output]);
 
+  const cover = useMemo(() => {
+    const maybeCover = output.blocks.find((block) => block.type === "image");
+    if (maybeCover) {
+      return maybeCover.data.file.url;
+    }
+  }, [output]);
+
   const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const form = e.currentTarget;
     const formData = new FormData(form);
     const data = Object.fromEntries(formData.entries());
 
     const body = {
-      cover: "https://via.placeholder.com/150",
+      cover: cover,
```

Comment:
Since the property name is the same as variable, usually, you can just put

```
body = {
   cover,
   // ....
}
```

Everything look good.